docs/docs-beta/docs/guides/non-python.md

```diff
@@ -1,73 +1,84 @@
+
+This guide covers how to run JavaScript with Dagster using Pipes, however, the same principle will apply to other languages.
 
 <details>
 <summary>Prerequisites</summary>
 
+To follow this guide, you'll need:
+
 - Familiarity with [Assets](/concepts/assets)
 - A basic understanding of JavaScript and Node.js
-</details>
 
-## Create a script using Tensorflow in JavaScript
+To run the examples, you'll need to install:
+
+- [Node.js](https://nodejs.org/en/download/package-manager/)
+- The following packages:
 
-{/* TODO: consider changing guide to step-based-guide template */}
+   ```bash
+   pip install dagster dagster-webserver tensorflow
+   ```
+</details>
 
-In this example, we will orchestrate a JavaScript script that reads in a CSV file, and uses the Tensorflow  to train a sequential model using Dagster Pipes.
+## Step 1: Create a script using Tensorflow in JavaScript
 
-Start by creating a file named `tensorflow/main.js` with the following contents:
+First, you'll create a JavaScript script that reads in a CSV file and uses Tensorflow to train a sequential model.
 
-<CodeExample filePath="guides/non-python/pipes-contrived-javascript.js" language="javascript" title="tensorflow/main.js" />
+Create a file named `tensorflow/main.js` with the following contents:
 
-## Create a Dagster asset that runs your script
+<CodeExample filePath="guides/non-python/pipes-contrived-javascript.js" language="javascript" title="tensorflow/main.js" />
 
-Create an asset that takes the `PipesSubprocessClient` resource, and set the `compute_kind` so that we know an alternate compute is being used for materialization. Then, use the resource to run your script using `node`.
+## Step 2: Create a Dagster asset that runs the script
 
-:::note
-You will need `node` in the environment you are running Dagster, along with Tensorflow installed.
-:::
+In Dagster, create an asset that:
 
-{/* TODO: consider adding instructions for installing dependencies */}
+- Uses the `PipesSubprocessClient` resource to run the script with `node`
+- Sets the `compute_kind` to `javascript`. This makes it easy to identify that an alternate compute will be used for materialization.
 
-<CodeExample filePath="guides/non-python/pipes-asset.py" language="python" title="Asset using Dagster Pipes." />
+<CodeExample filePath="guides/non-python/pipes-asset.py" language="python" />
 
-The stdout and stderr will be captured automatically and shown in the asset logs. If the command passed to Dagster Pipes returns a successful exit code, then an asset materialization result will be produced.
+When the asset is materialized, the stdout and stderr will be captured automatically and shown in the asset logs. If the command passed to Pipes returns a successful exit code, Dagster will produce an asset materialization result.
 
 ![Image of captured stdout](/img/placeholder.svg)
 
-## Send to, and receive data from, your script (Optional)
+## Step 3: Send and receive data from the script
```

Comment:
I removed the `optional` because it actually did feel like a requirement as Step 4 depends on this.